cpp/src/fil/fil.cu

```diff
@@ -305,7 +350,7 @@ struct forest {
         preds,
         num_values_to_transform,
         ot,
-        num_trees_ > 0 ? (1.0f / num_trees_) : 1.0f,
+        num_trees_ > 0 ? (real_t(1) / num_trees_) : real_t(1),
```

Comment:
In general, we should probably [prefer](https://herbsutter.com/2013/05/09/gotw-1-solution/) uniform initialization syntax (e.g. `real_t{1}`), but no need to go crazy about fixing every instance of it.

Reply:
I have a strong preference for `()` in constructor calls and constant conversions. I prefer to reserve `{}` to struct and array literals, as well as initializer lists.